src/main/scala/is/hail/expr/Type.scala

```diff
@@ -2031,6 +2034,9 @@ final case class TStruct(fields: IndexedSeq[Field]) extends Type {
   def isFieldDefined(region: MemoryBuffer, offset: Long, fieldIdx: Int): Boolean =
     !region.loadBit(offset, fieldIdx)
 
+  def isFieldDefined(region: Code[MemoryBuffer], offset: Code[Long], fieldIdx: Int): Code[Boolean] =
```

Comment:
Should the fieldIdx argument be `Code[Int]` to match the function you added above?

Reply:
I don't think we're ever checking the idx of the field at runtime.

Reply:
more on that: the generated code stuff doesn't currently care about the fieldIdx because e.g. all the offsets are stored as constants in the generated code to avoid having to allocate arrays to store them, so I think as it currently stands bad things can happen if the fieldIdx is stored as a Code[Int] instead of Int

Reply:
In my use cases `fieldIdx` is a "hail-expression-compile-time" constant so I use a Scala `Int`. I think that's the right choice, though I suppose everything would work just as well if `fieldIdx` was `Code[Int]`

Reply:
No, no, you can't access fields dynamically.  As @danking says, `fieldIdx` should always Hail compile time constant.